### ISSUE TITLE
📢 Azure/login v2 Release Announcement

### ISSUE THREAD

**Issue by MoChilia** [Azure/login#424]
👀 **What’s new in `azure/login@v2`?**

- The default runtime of `azure/login` is updated from `Node16` to `Node20` due to [the upcoming deprecation of Node16](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/).

⚠️ **This is breaking change**

- If you're using `azure/login` with a self-hosted runner, ensure it's running [Actions Runner image v2.308.0](https://github.com/actions/runner/releases/tag/v2.308.0) or later.

To apply this change, please update the version of `azure/login` to `v2` in your workflow files.
```yaml
    - name: Azure Login
      uses: azure/login@v2
      with:
        client-id: ${{ secrets.AZURE_CLIENT_ID }}
        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
        enable-AzPSSession: true
```

**Comment by JimSuplizio** [comment]
@MoChilia 
One thing that I'm noticing with the update to azure/login@v2 is a warning concerning the Az version that doesn't happen with the azure/login@v1. It's just an informational warning that's output during the `Pre Az CLI login` and I wanted to bring this to your attention.
Clearing azure cli accounts from the local cache.

/usr/bin/az account clear
{
  Result: '/usr/share/az_11.3.1/Az.Accounts/2.17.0/Az.Accounts.psd1',
  Success: true
}
Clearing azure powershell accounts from the local cache.
/usr/bin/pwsh -Command Clear-AzContext -Scope Process
/usr/bin/pwsh -Command Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
**WARNING: You're using Az version 11.3.1. The latest version of Az is 11.5.0. Upgrade your Az modules using the following commands:**
  Update-PSResource Az -WhatIf    -- Simulate updating your Az modules.
  Update-PSResource Az            -- Update your Az modules.

**Comment by YanaXu** [comment]
> @MoChilia One thing that I'm noticing with the update to azure/login@v2 is a warning concerning the Az version that doesn't happen with the azure/login@v1. It's just an informational warning that's output during the `Pre Az CLI login` and I wanted to bring this to your attention. Clearing azure cli accounts from the local cache.
> 
> /usr/bin/az account clear { Result: '/usr/share/az_11.3.1/Az.Accounts/2.17.0/Az.Accounts.psd1', Success: true } Clearing azure powershell accounts from the local cache. /usr/bin/pwsh -Command Clear-AzContext -Scope Process /usr/bin/pwsh -Command Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue **WARNING: You're using Az version 11.3.1. The latest version of Az is 11.5.0. Upgrade your Az modules using the following commands:** Update-PSResource Az -WhatIf -- Simulate updating your Az modules. Update-PSResource Az -- Update your Az modules.

@JimSuplizio, it's because the default Azure PowerShell version on the runner is not the latest. It's not related to the Azure Login Action release.

**Comment by JimSuplizio** [comment]
@MoChilia yes, but this warning wasn't happening on the v1 version, only the v2. I'm just bringing that to your attention.

**Comment by ChamalGomesHSO** [comment]
Hi - could you confirm if the issue https://github.com/Azure/login/issues/403 is expected to be resolved in v2 ? We just tried a CICD workflow run with v2 and we still get the errors message as specified in issue 403. i.e. We have concurrent jobs running with az login in both jobs, when one job finishes before another, the post clean up of credentials causes the other job to fail.

**Comment by MoChilia** [comment]
> @MoChilia can you please let me know if the `v2` tag will be updated to be the same commit as `v2.1.0`. It's still `v2.0.0`.

Hi @mennlo, we've updated `v2` to `v2.1.0`. Feel free to test it out now. Last week, we were testing the tag and awaiting customer feedback to ensure there are no issues with `v2.1.0`. Thanks for understanding.

**Comment by MoChilia** [comment]
> Hi - could you confirm if the issue #403 is expected to be resolved in v2 ? We just tried a CICD workflow run with v2 and we still get the error message as specified in issue 403. i.e. We have concurrent jobs running with az login in both jobs, when one job finishes before another, the post clean up of credentials causes the other job to fail. Keep in mind both jobs run on the same self hosted runner.

Hi @ChamalGomesHSO, the issue with #403 is due to installing azure-cli within the job, which can cause failures in the `pre` cleanup step. This has been fixed since `v1.6.1`. Your issue seems to be unrelated to that. 

It appears you have multiple runners on a single VM to run jobs in parallel, which can be risky. If you log in to different accounts in two jobs, you may switch to the wrong Azure account. We recommend using one single VM user per runner. For more details, see https://github.com/Azure/login?tab=readme-ov-file#security-hardening.

**Comment by ViltsuV** [comment]
Would be nice to have these kind of announcements in a changelog file or something.

**Comment by JackBruceShell** [comment]
<img width="1317" alt="Screenshot 2024-05-10 at 14 15 52" src="https://github.com/Azure/login/assets/102669576/a458fc46-2759-4ae7-a93d-ad78bcb5cf0c">

I am facing this issue when trying to change to azure/login@v2

Any help please?

**Comment by MoChilia** [comment]
Hi @JackBruceShell, could you open a new issue with your sample workflow and runner configuration?

**Comment by JackBruceShell** [comment]
> Hi @JackBruceShell, could you open a new issue with your sample workflow and runner configuration?

https://github.com/Azure/login/issues/442

**Comment by 19689188** [comment]
The completion is at stake with major illness and is asking for disability, in order to study , be disable and finally grasp the nation with dollar amount once completed with the board of education conduit is the name that went under, and now the board has HMI as NAvid has portrayed me and used Hypnosis motivational Institute in my behalf in order of violence of an accident. He has written for FCI for the logistics of the identity theft and also major jewelry theft, and Swiss money with major Obama named in the paragraphs, and I cannot recall what was said, but whatever was said was not good enough to partake the answer of the man himself. I am uncomfortable with not having a printer, and I have to make some changes and start printing the documentations that go out. That is all for now. 
    On Monday, April 6, 2026, 07:56:30 PM PDT, Yan Xu ***@***.***> wrote:   

 

Closed #424 as completed.
